src/core/include/iDynTree/Core/Rotation.h

```diff
@@ -271,6 +296,19 @@ namespace iDynTree
          */
         static Rotation Identity();
 
+        /**
+         * Construct a rotation matrix from the given quaternion representation
+         *
+         * The quaternion is expected to be ordered in the following way:
+         * - s \in \mathbb{R} the constituent of the quaterion
+         * - r \in \mathbb{R}^3 the imaginary part of the quaternion
+         *
+         * @param quaternion a quaternion representing a rotation
+         *
+         * @return The rotation matrix
+         */
+        static Rotation FromQuaternion(const iDynTree::Vector4& quaternion);
```

Comment:
I just noticed that this is a static method. Now I see, even if in the existing objects we tend to have actual methods called `from...` that modify the `*this` object.